File: Semestrul 3/MAP/lab 2/src/Sorters/BubbleSorter.java
package Sorters;

import java.util.List;

public class BubbleSorter extends AbstractSorter{
    public BubbleSorter(List<Integer> numbers) {
        super(numbers);
    }

    @Override
    public void sort() {
        for (int i = 0; i < numbers.size() - 1; i++) {
            for (int j = 0; j < numbers.size() - i - 1; j++) {
                if (numbers.get(j) > numbers.get(j + 1)) {
                    int temp = numbers.get(j);
                    numbers.set(j, numbers.get(j + 1));
                    numbers.set(j + 1, temp);
                }
            }
        }
    }
}

File: Semestrul 3/MAP/lab 10/sorters/BubbleSorter.cs
namespace lab10;

public class BubbleSorter(List<int> numbers) : AbstractSorter(numbers)
{
    public override void Sort()
    {
        for (var i = 0; i < numbers.Count - 1; i++)
        {
            for (var j = 0; j < numbers.Count - i - 1; j++)
            {
                if (numbers[j] > numbers[j + 1])
                {
                    (numbers[j], numbers[j + 1]) = (numbers[j + 1], numbers[j]);
                }
            }
        }
    }
}